migrator/migration.py

```diff
@@ -158,6 +175,46 @@ def create_bare_migrator_instance_in_org_space(self):
 
             if status == "failed":
                 raise Exception("Creation of migrator service instance failed.")
+
+            retries -= 1
+            time.sleep(config.SERVICE_CHANGE_POLL_TIME_SECONDS)
+
+        raise Exception("Checking migrator service instance timed out.")
+
+    def update_existing_cdn_domain(self):
+        params = {
+            "origin": self.origin_hostname,
+            "path": self.origin_path,
+            "forwarded_cookies": self.forwarded_cookies,
+            "forward_cookie_policy": self.forward_cookie_policy,
+            "forwarded_headers": self.forwarded_headers,
+            "insecure_origin": self.insecure_origin,
+            "error_responses": self.custom_error_responses,
+            "cloudfront_distribution_id": self.cloudfront_distribution_id,
+            "cloudfront_distribution_arn": self.cloudfront_distribution_arn,
+            "iam_server_certificate_name": self.iam_certificate_name,
+            "iam_server_certificate_id": self.iam_certificate_id,
+            "iam_server_certificate_arn": self.iam_certificate_arn,
+            "domain_internal": self.domain_internal,
+        }
+
+        cf.update_existing_cdn_domain_service_instance(
+            self.external_domain_broker_service_instance, params, self.client
+        )
+
+        retries = config.SERVICE_CHANGE_RETRY_COUNT
+
+        while retries:
+            status = cf.get_migrator_service_instance_status(
+                self.external_domain_broker_service_instance, self.client
+            )
+
+            if status == "succeeded":
+                return
+
+            if status == "failed":
+                raise Exception("Creation of migrator service instance failed.")
+
             retries -= 1
```

Comment:
I wonder if it's worth wrapping this logic in a function, since it's exactly the same as above

Reply:
That it does, good call out!

Reply:
I extracted this to its own method.